File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        vector<vector<int>> ans;
        int start=intervals[0][0];
        int end=intervals[0][1];
        int n=intervals.size();
        for(int i=1;i<n;i++){
            if(end<intervals[i][0]){
                ans.push_back({start, end});
                start=intervals[i][0];
                end=intervals[i][1];
            }
            else{
                end=max(end, intervals[i][1]);
            }
        }
        ans.push_back({start, end});
        return ans;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        arr=[]
        prev=intervals[0]

        for it in intervals:
            if it[0]<=prev[1]:
                prev[1]=max(prev[1], it[1])
            else:
                arr.append(prev)
                prev=it

        arr.append(prev)
        return arr 